invisible_cities/cities/components.py

```diff
@@ -168,20 +174,21 @@ def copy_mc_info(files_in : List[str], h5out: tb.File, event_numbers: List[int])
     file_out : tables.File
         The output h5 file.
     event_numbers : List[int]
-        List of event numbers for which the MC info is copied to the output file.
+        List of event numbers for which the MC info is copied
+        to the output file.
     """
 
-    writer = mcinfo_io.mc_info_writer(h5out)
+    writer = mcinfo_io.mc_writer(h5out)
 
     for f in files_in:
-        with tb.open_file(f, "r") as h5in:
-            try:
-                event_numbers_in_file = h5in.root.MC.extents.cols.evt_number[:]
-                event_numbers_to_copy = list(evt for evt in event_numbers \
-                                             if evt in event_numbers_in_file)
-                mcinfo_io.copy_mc_info(h5in, writer, event_numbers_to_copy)
-            except tb.exceptions.NoSuchNodeError:
-                continue
+        try:
+            event_numbers_in_file = mcinfo_io.get_event_numbers_in_file(f)
+            event_numbers_to_copy = list(evt for evt in event_numbers \
+                                         if evt in event_numbers_in_file)
+            mcinfo_io.copy_mc_info(f, writer, event_numbers_to_copy,
+                                   db_file, run_number)
+        except tb.exceptions.NoSuchNodeError:
```

Comment:
This is confusing, when do we expect NoSuchNodeError? Does this mean we skip a file that does not have MC information? This doesnt seem right... Also there is no protection that all `event_numbers` are eventually contained in the output file

Reply:
On the other hand we never check that all of the `event_numbers` are actually copied, guess we could append `event_numbers_to_copy` in a list and at the end compare that its the same as the `event_numbers`list?